packages/realm-react/src/useAuthOperation.tsx

```diff
@@ -0,0 +1,72 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { useCallback } from "react";
+import { AuthError, AuthOperationName, OperationState } from "./types";
+import { useAuthResult } from "./AppProvider";
+
+export function useAuthOperation<Args extends unknown[], Result>({
+  operation,
+  operationName,
+  onSuccess = () => undefined,
+}: {
+  operation: (...args: Args) => Promise<Result | void>;
+  operationName: AuthOperationName;
+  onSuccess?: (...args: Args) => void;
+}) {
+  const [result, setResult] = useAuthResult();
+
+  return useCallback<(...args: Args) => ReturnType<typeof operation>>(
```

Comment:
I wouldn't expect this to have any (positive) effect, since the `operation` is not memoized from the caller, it will invalidate this memoization on every render.

Reply:
You may be right.  I'll have to do a deeper dive into `useCallback`.  I may just be being defensive here.

Reply:
Good catch.  I'll update the callers so that they provide a memoized operation to this.